torch/_dynamo/eval_frame.py

```diff
@@ -355,11 +378,11 @@ def __enter__(self):
                 "to use torch._dynamo.optimize(...) as an annotation/decorator. "
             )
         self.cleanup_fns = [enter() for enter in self.enter_exit_hooks]
-        self.prior = _maybe_set_eval_frame(self.callback)
+        self.prior = _maybe_set_eval_frame(self.callback, id(self), "enter")
```

Comment:
What's the lifetime of a _TorchDynamoContext? Is it the entire lifetime of the program or do we need to worry about `id(self)` being reused?

Reply:
it gets created and deleted within the torch.compile wrapping call, so yes there could be a key conflict